### ISSUE TITLE
Add client-side input validation for better UX

### ISSUE THREAD

**Issue by luke3butler** [OrasesWPDev/handy-custom#15]
**⚠️ Note: This is a potential issue identified during code review that needs investigation. There may be specific reasons for the current implementation approach.**

## Problem
No client-side validation before AJAX requests, relying entirely on server-side validation.

## Current Behavior
- All validation happens on the server
- Users must wait for server response to see validation errors
- Invalid requests sent to server unnecessarily
- No immediate feedback for user input errors

## Benefits of Client-Side Validation
- Immediate user feedback without server round-trip
- Reduced server requests for obviously invalid data
- Better user experience with instant error messages
- Reduced server load from invalid requests
- Faster perceived performance

### 1. Validate Filter Values
```javascript
function validateFilters(filters) {
    var validCategories = window.handyCustomAjax.validCategories || [];
    
    if (filters.category && \!validCategories.includes(filters.category)) {
        showError('Invalid category selected');
        return false;
    }
    
    return true;
}
```

### 2. Localize Validation Data
```php
// In PHP - provide validation data to JavaScript
wp_localize_script('handy-custom-products', 'handyCustomAjax', array(
    'ajaxUrl' => admin_url('admin-ajax.php'),
    'nonce' => wp_create_nonce('handy_custom_nonce'),
    'validCategories' => array_keys($categories),
    'validGrades' => array_keys($grades),
    // ... other validation data
));
```

### 3. User-Friendly Error Messages
```javascript
function showValidationError(message) {
    $('.validation-errors').html('<p class="error">' + message + '</p>').show();
    setTimeout(function() {
        $('.validation-errors').fadeOut();
    }, 5000);
}
```

### 4. Input Sanitization
```javascript
function sanitizeFilterValue(value) {
    return value.replace(/[^a-zA-Z0-9\-_]/g, '');
}
```

## Important Notes
- Client-side validation is for UX only
- Server-side validation must remain as security layer
- Should not replace existing security measures
- Focus on obvious input errors and user guidance

## Acceptance Criteria
- [ ] Client-side validation for all filter inputs
- [ ] User-friendly error messages with clear guidance
- [ ] Validation data properly localized from PHP
- [ ] Reduced invalid AJAX requests (measurable)
- [ ] Server-side validation remains unchanged
- [ ] No security weakening from client-side changes

## Priority
Low - User experience enhancement, not critical functionality

**Labels:** enhancement, frontend, user-experience, validation
